client/local/specio/job_spec_io.go

```diff
@@ -23,7 +23,8 @@ type jobSpecReadWriter struct {
 	referenceSpecFileName   string
 	referenceAssetDirName   string
 
-	specFS afero.Fs
+	specFS           afero.Fs
+	pathToSpecsCache map[string][]*model.JobSpec
```

Comment:
Why are we adding a cache. ?

Reply:
in apply, reading is happening in multiple places ([here](https://github.com/goto/optimus/pull/221/files#diff-4392d72697b8f4531bbf475d1d12b598fa90171e14af6fe0b2dd7f51a8cabab8R384), [here](https://github.com/goto/optimus/pull/221/files#diff-4392d72697b8f4531bbf475d1d12b598fa90171e14af6fe0b2dd7f51a8cabab8R360), [here](https://github.com/goto/optimus/pull/221/files#diff-4392d72697b8f4531bbf475d1d12b598fa90171e14af6fe0b2dd7f51a8cabab8R425), [here](https://github.com/goto/optimus/pull/221/files#diff-4392d72697b8f4531bbf475d1d12b598fa90171e14af6fe0b2dd7f51a8cabab8R458), [here](https://github.com/goto/optimus/pull/221/files#diff-4392d72697b8f4531bbf475d1d12b598fa90171e14af6fe0b2dd7f51a8cabab8R483), and [here](https://github.com/goto/optimus/pull/221/files#diff-4392d72697b8f4531bbf475d1d12b598fa90171e14af6fe0b2dd7f51a8cabab8R525))
better to use cache to prevent re-reading the spec